lib/internal/debugger/inspect_repl.js

```diff
@@ -961,7 +962,13 @@ function createRepl(inspector) {
       },
 
       takeHeapSnapshot(filename = 'node.heapsnapshot') {
-        return new Promise((resolve, reject) => {
+        if (heapSnapshotPromise) {
+          print(
+            'Cannot take heap snapshot because another snapshot is in progress.'
+          );
+          return heapSnapshotPromise;
+        }
+        heapSnapshotPromise = new Promise((resolve, reject) => {
```

Comment:
do we have a tarde-off between printing the message versus silently returning the active one?

Reply:
ok, we are in debugging mode already, so printing makes sense to me!